sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_models.py

```diff
@@ -48,11 +48,70 @@ class SettingSelector:
     :keyword label_filter: A filter to select configuration settings based on their labels. Default is value is
      EMPTY_LABEL i.e. (No Label) as seen in the portal.
     :type label_filter: Optional[str]
+    :keyword tag_filters: A filter to select configuration settings based on their tags. This is a list of strings
+     that will be used to match tags on the configuration settings. Reserved characters (\\*, \\, ,) must be escaped
+     with backslash if they are part of the value.
+    :type tag_filters: Optional[List[str]]
     """
 
-    def __init__(self, *, key_filter: str, label_filter: Optional[str] = EMPTY_LABEL):
+    def __init__(
+        self, *, key_filter: str, label_filter: Optional[str] = EMPTY_LABEL, tag_filters: Optional[List[str]] = None
+    ):
+        if tag_filters is not None:
+            if not isinstance(tag_filters, list):
+                raise TypeError("tag_filters must be a list of strings.")
+            if len(tag_filters) > 5:
+                raise ValueError("tag_filters cannot be longer than 5 items.")
+            for tag in tag_filters:
+                if not tag:
+                    raise ValueError("Tag filter cannot be an empty string or None.")
+                if not isinstance(tag, str) or "=" not in tag or tag.startswith("="):
+                    raise ValueError("Tag filter " + tag + ' does not follow the format "tagName=tagValue".')
+                # Check for unescaped reserved characters
+                self._validate_tag_filter_escaping(tag)
+
         self.key_filter = key_filter
         self.label_filter = label_filter
+        self.tag_filters = tag_filters
+
+    def _validate_tag_filter_escaping(self, tag: str) -> None:
```

Comment:
We dont need to validate this on client side. Server will validate and throw 400 if there are invalid characters

Reply:
We skipped these validations in the .NET provider because the client cannot cover all the possible invalid character checks that server has. And we have precedent for not validating key/labels on the client side. We do a basic validation (like presence of `=` character and non-empty tag key). That's sufficient for client side.